src/hash.cr

```diff
@@ -272,6 +272,26 @@ class Hash(K, V)
     self
   end
 
+  def select(&block : K, V -> U)
+    inject({} of K => V) do |memo, k, v|
+      yield(k, v) ? memo.merge!({k => v}) : memo
+    end
+  end
+
+  def select!(&block : K, V -> U)
+    delete_if{|k, v| !yield k, v }
```

Comment:
The "I like" and "I don't like" is essentially the problem that a style guide solves.

For now, until there is an official guide, for this case it is simple enough to look at the general coding style used in the project (search code for `map {` vs `map{` etc). Which is clearly in favour of `delete_if { |k, v| !yield k, v }`, that is space both before `{` and `|`.

Regarding taking things from bbatsov as a base isn't necessarily a bad idea. I've yet to see any ruby style guide that isn't more or less a subset of what's in there. A lot of what it does is mention the different ways things can be done and then argue that you should be choosing one consistent style for your own project.